experiments/tf_trainer/common/tfrecord_input.py

```diff
@@ -56,9 +56,12 @@ def text_feature(self) -> str:
     return self._text_feature
 
   def train_input_fn(self) -> types.FeatureAndLabelTensors:
-    """input_fn for TF Estimators for training set."""
+    """input_fn for TF Estimators for training set.
+   
+    Automatically repeats over input data forever.
+    """
     assert FLAGS.train_path
-    return self._input_fn_from_file(FLAGS.train_path)
+    return self._input_fn_from_file(FLAGS.train_path).repeat()
```

Comment:
Should the .repeat() also be on the validate_input_fn()?

Reply:
No, I think to denoise the graphs we should always use the same portion of the validation set - or, actually, the entire validation set each step.

Reply:
SGTM.